Source/UI/NewUI/PaymentForm/View/PaymentHeaderView.swift

```diff
@@ -71,14 +72,10 @@ public final class PaymentHeaderView: UIView {
   }
 
   private func updateIconsStack() {
-    guard let icons = style?.schemeIcons else {
-      iconsStackView.isHidden = true
-      return
-    }
-    iconsStackView.isHidden = false
-    for image in icons where image != nil {
+    iconsStackView.isHidden = style?.isIconsViewHidden ?? true
+    for scheme in supportedSchemes {
       let imageView = UIImageView().disabledAutoresizingIntoConstraints()
-      imageView.image = image?.withRenderingMode(.alwaysOriginal)
+      imageView.image = Constants.Bundle.SchemeIcon(scheme: scheme).image?.withRenderingMode(.alwaysOriginal)
```

Comment:
You really should remove all the images that exist in the stack **before** adding new ones. I know you mentioned this list is not expected to be updated (_yet_), but if your lifecycle calls the same method multiple times (ie: even simple things like scrolling in & out of view), you end up keep adding images.